common/app/routes/challenges/components/Show.jsx

```diff
@@ -72,25 +79,25 @@ export class Challenges extends PureComponent {
     updateTitle: PropTypes.func.isRequired
   };
 
-  componentWillMount() {
-    this.props.updateTitle(this.props.title);
-  }
-
   componentDidMount() {
     if (!this.props.areChallengesLoaded) {
       this.props.fetchChallenges();
     }
+    if (this.props.title) {
```

Comment:
Without this check, `undefined` was being sent with `updateTitle` making `title` `'Learn to code | Free Code Camp'` after the server markup was rendered and react took over.

But there is a need to update `title` when coming from a non-react page like the front page or `about`. Title is not undefined in this case.